lib/Runtime/Library/JavascriptProxy.h

```diff
@@ -54,6 +54,19 @@ namespace Js
         static BOOL Is(_In_ RecyclableObject* obj);
         static JavascriptProxy* FromVar(Var obj) { AssertOrFailFast(Is(obj)); return static_cast<JavascriptProxy*>(obj); }
         static JavascriptProxy* UnsafeFromVar(Var obj) { Assert(Is(obj)); return static_cast<JavascriptProxy*>(obj); }
+
+        // before recursively calling something on 'target' use this helper in case there is nesting of proxies.
+        // the proxies could be deep nested and cause SO when processed recursively.
+        static const JavascriptProxy* UnwrapNestedProxies(const JavascriptProxy* proxy)
+        {
+            while (proxy->handler != nullptr && JavascriptProxy::Is(proxy->target))
+            {
+                proxy = JavascriptProxy::FromVar(proxy->target);
+            }
+
+            return proxy;
+        }
+
```

Comment:
non-trivial enough to be implemented in the cpp file?

Reply:
I think it is simple enough. We have more complex methods in this file.

---
In reply to: [205296033](https://github.com/Microsoft/ChakraCore/pull/5529#discussion_r205296033) [](ancestors = 205296033)